buildscripts/cmake/SetupBuildEnvironment.cmake

```diff
@@ -10,7 +10,7 @@ set(CMAKE_UNITY_BUILD_BATCH_SIZE 12)
 if (CC_IS_GCC)
     message(STATUS "Using Compiler GCC ${CMAKE_CXX_COMPILER_VERSION}")
 
-    set(CMAKE_CXX_FLAGS_DEBUG   "-g")
+    set(CMAKE_CXX_FLAGS_DEBUG   "-g -Wall -Wextra -Wfloat-equal")
```

Comment:
This should perhaps be in `SetupCompilerWarnings.cmake` instead

Reply:
Moved there, I don't think it'll stay there for long though.

Reply:
If it is to stay, we'd need to disable it for quite a few thirdparty packages
And maybe eve for some of our code

Reply:
Like opus, fluidsynth, flac, harfbuzz, pugixml, dr_libs, picojson and beatroot